src/payment/strategies/laybuy/laybuy-payment-strategy.spec.ts

```diff
@@ -0,0 +1,167 @@
+
+        jest.spyOn(store, 'dispatch');
+
+        jest.spyOn(formPoster, 'postForm')
+            .mockImplementation((_url, _data, callback = () => {}) => callback());
+
+        jest.spyOn(orderActionCreator, 'finalizeOrder')
+            .mockReturnValue(finalizeOrderAction);
+
+        jest.spyOn(orderActionCreator, 'submitOrder')
+            .mockReturnValue(submitOrderAction);
+
+        jest.spyOn(paymentActionCreator, 'submitPayment')
+            .mockReturnValue(submitPaymentAction);
+
+        strategy = new LaybuyPaymentStrategy(
+            store,
+            orderActionCreator,
+            paymentActionCreator,
+            formPoster
+        );
+    });
+
+    describe('#execute()', () => {
+        it('submits order without payment data', async () => {
+            const payload = getOrderRequestBody();
+            const options = { methodId: 'laybuy' };
+
+            await strategy.execute(payload, options);
+
+            expect(orderActionCreator.submitOrder).toHaveBeenCalledWith(omit(payload, 'payment'), options);
+            expect(store.dispatch).toHaveBeenCalledWith(submitOrderAction);
+        });
+
+        it('returns checkout state', async () => {
+            const output = await strategy.execute(getOrderRequestBody());
+
+            expect(output).toEqual(store.getState());
+        });
+
+        it('returns checkout state with undefined payload', () => {
+            const payload = {
+                payment: undefined,
+            };
+            expect(() => strategy.execute(payload))
+                .toThrow(PaymentArgumentInvalidError);
+        });
+
+        it('redirect to Laybuy if additional action is required', async () => {
+            const error = new RequestError(getResponse({
+                ...getErrorPaymentResponseBody(),
+                errors: [
+                    { code: 'additional_action_required' },
+                ],
+                provider_data: {
+                    redirect_url: 'https://sandbox-payment.laybuy..com',
+                },
+                status: 'error',
+            }));
+
+            jest.spyOn(paymentActionCreator, 'submitPayment')
+                .mockReturnValue(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, error)));
+
+            strategy.execute(getOrderRequestBody());
+
+            await new Promise(resolve => process.nextTick(resolve));
+
+            expect(formPoster.postForm).toHaveBeenCalledWith('https://sandbox-payment.laybuy..com', {});
+        });
+
+        it('reject payment when error is different to additional_action_required', async () => {
+            const response = new RequestError(getResponse(getErrorPaymentResponseBody()));
+
+            jest.spyOn(paymentActionCreator, 'submitPayment')
+                .mockReturnValue(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, response)));
+
+            try {
+                await strategy.execute(getOrderRequestBody());
```

Comment:
avoid `try/catch`, because if the exception is never thrown, no expectations would be called. Test could silently break.
```
expect(async () => await strategy.execute()).toThrowError(RequestError)
```